src/platform/ti/cc13xx_26xx/BLEManagerImpl.cpp

```diff
@@ -179,21 +266,48 @@ CHIP_ERROR BLEManagerImpl::_GetDeviceName(char * buf, size_t bufSize)
     return ret;
 }
 
-CHIP_ERROR BLEManagerImpl::_SetDeviceName(const char * deviceName)
+uint8_t BLEManagerImpl::getDeviceName(char * buf, size_t bufSize)
 {
-    CHIP_ERROR ret = CHIP_NO_ERROR;
+    uint8_t ret = FAILURE;
+    if (bufSize <= GAP_DEVICE_NAME_LEN)
+    {
+        Platform::CopyString(buf, bufSize, mDeviceName);
+    }
+    else
+    {
+        ret = SUCCESS;
+    }
+    return ret;
```

Comment:
This looks a bit strange, isn't the if the wrong way around and also in case it actually does the copy, wouldn't it still return FAILURE?

Reply:
Thanks for catching this, a fix has been pushed